include/watchdog.h

```diff
@@ -1,116 +1,206 @@
 /*
- * Copyright (c) 2015 Intel Corporation.
+ * Copyright (c) 2017 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file
+ * @brief Public API for watchdog drivers.
+ */
+
+#if defined(CONFIG_WDT_LEGACY_API)
+
+/*
+ * This is the default, and will be the way until the new API below
+ * will be enforced everywhere.
+ */
+#include <watchdog_legacy.h>
+
+#else
+
 #ifndef _WDT_H_
 #define _WDT_H_
 
 #include <zephyr/types.h>
-#include <device.h>
 #include <misc/util.h>
+#include <device.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#define WDT_MODE		(BIT(1))
-#define WDT_MODE_OFFSET         (1)
-#define WDT_TIMEOUT_MASK        (0xF)
+/**
+ * @brief Enable calling application callback function on watchdog timeout.
+ */
+#define WDT_MODE_CALLBACK	BIT(0)
 
-enum wdt_mode {
-	WDT_MODE_RESET = 0,
-	WDT_MODE_INTERRUPT_RESET
+/**
+ * @brief Pause watchdog timer when CPU is in sleep state.
+ */
+#define WDT_PAUSE_IN_SLEEP	BIT(1)
```

Comment:
Devil's advocacy: why is this an option?  Under what circumstance would you *want* a driver to fail to automatically pause a watchdog during sleep?

Maybe what we really want is a flag set by the driver that tells the user "watchdog does NOT automatically pause for sleep, you have to do it yourself!" ?

Reply:
@andyross  I am sorry, I am not sure what you mean, since when driver controls the user not the other way.
It is straightforward I guess, you either want watchdog counter to stop counting when CPU is in sleep state or you want it to keep counting, it depends on the application.